perfkitbenchmarker/vm_util.py

```diff
@@ -501,3 +504,51 @@ def GenerateSSHConfig(vms):
     ofp.write(template.render({'vms': vms}))
   logging.info('ssh to VMs in this benchmark by name with: '
                'ssh -F {0} <vm name>'.format(target_file))
+
+
+@contextlib.contextmanager
+def RunDStatIfConfigured(vms):
+  """Installs and runs dstat on 'vms' if FLAGS.dstat is set.
+
+  On exiting the context manager, the results are copied to the run temp dir.
+
+  Args:
+    vms: List of virtual machines.
+
+  Yields:
+    None
+  """
+  if not FLAGS.dstat:
+    yield
+    return
+
+  lock = threading.Lock()
+  pids = {}
+  file_names = {}
+
+  def StartDStat(vm):
+    vm.Install('dstat')
+    dstat_file = os.path.join(VM_TMP_DIR, '{0}-dstat.csv'.format(vm.name))
```

Comment:
Are you getting one dstat file per VM?  Per Benchmark?

On Tue, May 26, 2015 at 9:57 AM, Connor McCoy <notifications@github.com>
wrote:

> In perfkitbenchmarker/vm_util.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/282#discussion_r31055176>
> :
>
> > +    vms: List of virtual machines.
> > +
> > +  Yields:
> > +    None
> > +  """
> > +  if not FLAGS.dstat:
> > +    yield
> > +    return
> > +
> > +  lock = threading.Lock()
> > +  pids = {}
> > +  file_names = {}
> > +
> > +  def StartDStat(vm):
> > +    vm.Install('dstat')
> > +    dstat_file = os.path.join(VM_TMP_DIR, '{0}-dstat.csv'.format(vm.name))
>
>  I think the filename should be a parameter ... or at least part of the
> filename. The way you have this right now the user has no control over the
> name. If they start and stop a couple of times it looks like they will only
> get the latest.
>
> True, but only if they re-use the same run URI. I can add an additional
> URI to the file name, or a user-specified suffix, but since each VM will
> create a local file, a single file name provided by the user won't work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/282/files#r31055176>
> .
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/